File: api-posts/src/main/java/com/github/alexeysol/geekregimeapiposts/configurations/IndexerConfiguration.java
package com.github.alexeysol.geekregimeapiposts.configurations;

import com.github.alexeysol.geekregimeapiposts.utils.search.Indexer;
import org.springframework.boot.ApplicationArguments;
import org.springframework.boot.ApplicationRunner;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class IndexerConfiguration {
    @Bean
    public ApplicationRunner buildIndex(Indexer indexer) {
        return (ApplicationArguments args) ->
            indexer.indexPersistedData("com.github.alexeysol.geekregimeapiposts.models.entities.Post");
    }
}

File: api-users/src/main/kotlin/com/github/alexeysol/geekregimeapiusers/configurations/IndexerConfiguration.kt
package com.github.alexeysol.geekregimeapiusers.configurations

import com.github.alexeysol.geekregimeapiusers.utils.search.Indexer
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class IndexerConfiguration {
    @Bean
    fun buildIndex(indexer: Indexer): ApplicationRunner {
        return ApplicationRunner {
            indexer.indexPersistedData("com.github.alexeysol.geekregimeapiusers.models.entities.User")
        }
    }
}
